### ISSUE TITLE
BlockStateMeta changes not persisting 

### ISSUE THREAD

**Issue by abandonedaccount6235** [PaperSpigot/Paper#10811]
### Expected behavior

I have [a command](https://github.com/Pikachu920/blockstatemeta-issue/blob/main/src/main/java/io/github/pikachu920/blockstatemetaissue/DemonstrateBugCommand.java#L18) which gives the executor a glowing sign and broadcasts the glowing state of the sign. I expected the sign to be glowing when given and the broadcasts to say the sign is glowing.

```
> demo
[16:42:45 INFO]: Is the front glowing? true
[16:42:45 INFO]: Is the back glowing? true
```

### Observed/Actual behavior

The sign was not glowing when given to the player and the broadcasts say the sign is not glowing.
```
> demo
[16:44:42 INFO]: Is the front glowing? false
[16:44:42 INFO]: Is the back glowing? false
```

### Steps/models to reproduce

I have created a test plugin available at https://github.com/Pikachu920/blockstatemeta-issue. You can build that plugin and run the `demo` command (either in game or from console). The glowing state of the sign will be broadcast and, if in game, you will be given the sign. The output should match the observed behavior section.

### Plugin and Datapack List

Just the test plugin mentioned before
```
[16:43:11 INFO]: Server Plugins (1):
[16:43:11 INFO]: Bukkit Plugins:
[16:43:11 INFO]:  - blockstatemeta-issue
```
```
[16:43:41 INFO]: There are 3 data pack(s) enabled: [vanilla (built-in)], [file/bukkit (world)], [paper (built-in)]
[16:43:41 INFO]: There are no more data packs available
```

### Paper version

```
[17:01:17 INFO]: This server is running Paper version 1.20.6-115-master@9d6f2cc (2024-05-28T14:55:16Z) (Implementing API version 1.20.6-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.20.6-103-4e10fad (MC: 1.20.6)
```

### Other

This seems to have been introduced by https://github.com/PaperMC/paper/commit/4e10fadfdc8f854feeabf916c23a71e21fb0b29f (build 103). Build 102 works as expected.

**Comment by Machine-Maker** [comment]
Can you try the build that is linked at the top of this PR and see if it fixes your issue? <https://github.com/PaperMC/Paper/pull/10812>

**Comment by Pikachu920** [comment]
> Can you try the build that is linked at the top of this PR and see if it fixes your issue? #10812

yes, it does. thank you

```
> demo
[19:48:15 INFO]: Is the front glowing? true
[19:48:15 INFO]: Is the back glowing? true
> ver
[19:48:19 INFO]: Checking version, please wait...
[19:48:20 INFO]: This server is running Paper version 1.20.6-DEV-HEAD@89fff1d (2024-05-28T23:37:40Z) (Implementing API version 1.20.6-R0.1-SNAPSHOT)
```
